### PR TITLE
K8s secret backup improvements

### PR DESCRIPTION
**Pull request by ctyano**
# Description

_description_with_details_and_reasoning_

## Type of change

- [x] Bug fix
- [x] New feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

## Related issue/PR

**Delete this section if there are no issues or pull requests that relate to this pull request.**
- Fixes #_issue_
- Closes #_PR_

---

## Checklist

- [ ] Followed the guidelines in the CONTRIBUTING document
- [ ] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [ ] Tested and linted the code
- [ ] Commented the code
- [ ] Made corresponding changes to the documentation
- [ ] Passed all pipeline checking

## Checklist for maintainer
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge

### CODE REVIEW

File: pkg/identity/certificated.go [hunk 1]
```diff
@@ -82,6 +82,42 @@ func Certificated(idConfig *IdentityConfig, stopChan <-chan struct{}) error {
 		return w.Save()
 	}
 
+	identityProvisioningRequest := func(idConfig *IdentityConfig, handler *identityHandler, forceInit bool) (err error, identity *InstanceIdentity, keyPem []byte) {
```
Comment (mlajkim) [inline pkg/identity/certificated.go:85]: @ctyano I was wondering if we should have a log message with `forceInit` value to differentiate between normal and unusual call of the function`identityProvisioningRequest`

`if forceInit { ... message ... } ` would be enough? Would like to hear your opinion.

Comment (mlajkim) [reply pkg/identity/certificated.go:85]: was handled here:
https://github.com/AthenZ/k8s-athenz-sia/pull/12/commits/4f07ea168903882a57a509b2e8439e3fb5432976

File: pkg/identity/certificated.go [hunk 2]
```diff
@@ -82,6 +82,91 @@ func Certificated(idConfig *IdentityConfig, stopChan <-chan struct{}) (error, <-
...
+				log.Infof("Successfully saved x509 certificate to kubernetes secret")
+			} else {
+				log.Debugf("Skipping to save x509 certificate temporary backup to Kubernetes secret[%s]", idConfig.CertSecret)
+			}
+		}
+
+		return
+	}
+
+	roleCertProvisioningRequest := func(idConfig *IdentityConfig, handler *identityHandler) (roleCerts [](*RoleCertificate)) {
```
Comment (mlajkim) [inline pkg/identity/certificated.go:119]: Do you think we need this
`idConfig *IdentityConfig, handler *identityHandler`?

Seems like we don't as it is defined above anyway.

File: pkg/identity/certificated.go [hunk 3]
```diff
@@ -97,53 +182,29 @@ func Certificated(idConfig *IdentityConfig, stopChan <-chan struct{}) (error, <-
...
+				log.Errorf("Failed to retrieve x509 certificate from identity provider: %s", err.Error())
 			}
 		}
 
 		if identity == nil || len(keyPem) == 0 {
 			if idConfig.CertSecret != "" && strings.Contains(idConfig.Backup, "read") {
 				log.Infof("Attempting to load x509 certificate temporary backup from kubernetes secret[%s]...", idConfig.CertSecret)
 
-				identity, keyPem, err = handler.GetX509CertFromSecret()
+				backupIdentity, backupKeyPem, err = handler.GetX509CertFromSecret()
```
Comment (ctyano) [inline pkg/identity/certificated.go:198]: @mlajkim Should we use `tempIdentity` and `tempKeyPem` instead?

File: pkg/identity/certificated.go (hunk 2 above)
Comment (mlajkim) [reply pkg/identity/certificated.go:119]: @ctyano Does this apply to the `identityProvisioningRequest` too?
Will handle them all later after you review the logic part.

Comment (mlajkim) [reply pkg/identity/certificated.go]: @ctyano handled!

**Review by WindzCUHK — APPROVED** [review]

**Pull request merged**